frontend/pages/ManageControlsPage/OSUpdates/components/CurrentVersionSection/CurrentVersionSection.tsx

```diff
@@ -29,18 +30,47 @@ export type IFilteredOperatingSystemVersion = Omit<
 const baseClass = "os-updates-current-version-section";
 
 interface ICurrentVersionSectionProps {
+  router: InjectedRouter;
   currentTeamId: number;
+  queryParams: ReturnType<typeof parseOSUpdatesCurrentVersionsQueryParams>;
 }
 
+const DEFAULT_SORT_DIRECTION = "desc";
+const DEFAULT_SORT_HEADER = "hosts_count";
+const DEFAULT_PAGE = 0;
+const DEFAULT_PAGE_SIZE = 8;
+
+export const parseOSUpdatesCurrentVersionsQueryParams = (queryParams: {
+  page?: string;
+  order_key?: string;
+  order_direction?: "asc" | "desc";
+}) => {
+  const sortHeader = queryParams?.order_key ?? DEFAULT_SORT_HEADER;
+  const sortDirection = queryParams?.order_direction ?? DEFAULT_SORT_DIRECTION;
+  const page = queryParams?.page
+    ? parseInt(queryParams.page, 10)
+    : DEFAULT_PAGE;
+  const pageSize = DEFAULT_PAGE_SIZE;
+
+  return {
+    page,
+    order_key: sortHeader,
+    order_direction: sortDirection,
+    per_page: pageSize,
+  };
+};
+
 const CurrentVersionSection = ({
+  router,
   currentTeamId,
+  queryParams,
 }: ICurrentVersionSectionProps) => {
   const { data, isError, isLoading: isLoadingOsVersions } = useQuery<
     IOSVersionsResponse,
     AxiosError
   >(
-    ["os_versions", currentTeamId],
-    () => getOSVersions({ teamId: currentTeamId }),
+    ["os_versions", currentTeamId, queryParams],
+    () => getOSVersions({ teamId: currentTeamId, ...queryParams }),
```

Comment:
We don't need to hold up the PR for this, but whenever query key array starts getting getting longer than two or three elements, I like to switch to the pattern where the query key is a single-element array that contains an object with a predefined shape. Some recent examples are /pages/ManageControlsPage/Scripts/Scripts.tsx and /pages/hosts/details/cards/Software/SelfService/SelfService.tsx.


The reason is that the order of the array elements matters to react-query when trying to match keys in the cache. So we get more predictable caching with the single-element array pattern by eliminating the need to worry about what order we are inserting elements into the array.